### PR TITLE
major release reqired, not fully compat

### PR DESCRIPTION
**Pull request by earonesty**
this uses notanorm 3.X (sqlglot) for ddl instead of sqlite parsing for non-sqlite db's

### CODE REVIEW

File: omen2/codegen.py [hunk 1]
```diff
@@ -108,13 +108,13 @@ def __init__(self, id, color: str = "green"):
...
-            if col.default or not col.notnull:
+            if col.default is not None or not col.notnull:
                 # derive default value from the db default value
-                if col.default:
+                if col.default is not None:
                     try:
                         # check valid python
                         defval = pytype(str(col.default))  # pylint: disable=eval-used
-                    except (ValueError, NameError):
+                    except (ValueError, NameError, TypeError):
```
Comment (higleyc) [inline omen2/codegen.py:117]: when does this arise?

File: omen2/types.py [hunk 2]
```diff
@@ -34,9 +34,11 @@ def bool_type(arg):
 
 def string_type(arg):
     """Convert sql string to str, this function must have the __name__ 'str'"""
-    assert arg[0] in ("'", '"')
-    # return with quotes
-    return arg
+    if arg[0] in ("'", '"'):
+        return arg
+    else:  # pragma: no cover
```
Comment (higleyc) [inline omen2/types.py:39]: seems like a bug paver

File: omen2/codegen.py (hunk 1 above)
Comment (earonesty) [reply omen2/codegen.py:117]: if the default is some weird unexpected thing, like strftime("%s"), that doesn't translate to a python default

File: omen2/types.py (hunk 2 above)
Comment (earonesty) [reply omen2/types.py:39]: sqlglot doesn't normalize defaults in any way.   so we're kinda left to our own devices on that

Comment (earonesty) [reply omen2/types.py:39]: probably going to save that for after mssql support, which i think is more important

Comment (earonesty) [reply omen2/types.py:39]: i'll try to write a test for this

File: omen2/codegen.py (hunk 1 above)
Comment (earonesty) [reply omen2/codegen.py:117]: ok traced it down.  in sqlite, a type can be different than a default.   you can have INT DEFAULT('yo').   we apparently rely on this, but i think it's ok to remove it and fix.

File: omen2/types.py (hunk 2 above)
Comment (earonesty) [reply omen2/types.py:39]: same issue as above, in sqlite if you have a default (4) for a text column, you can get this.   again, think its ok to remove

File: omen2/codegen.py (hunk 1 above)
Comment (earonesty) [reply omen2/codegen.py]: removed, we'll see if we needed it

Comment (higleyc) [reply omen2/codegen.py]: yikes interesting. Okay sounds good

File: omen2/types.py (hunk 2 above)
Comment (higleyc) [reply omen2/types.py]: okay understood

**Review by higleyc — APPROVED** [review]

**Pull request merged**